studio/pages/project/[ref]/logs-explorer/saved.tsx

```diff
@@ -9,29 +9,33 @@ import LogsExplorerLayout from 'components/layouts/LogsExplorerLayout/LogsExplor
 
 import Table from 'components/to-be-cleaned/Table'
 import { useRouter } from 'next/router'
+import { UserContent } from 'types'
 
-export const LogsExplorerPage: NextPage = () => {
+export const LogsSavedPage: NextPage = () => {
   const { content, ui } = useStore()
   const router = useRouter()
   const { ref } = router.query
 
   useEffect(() => {
-    content.load()
+    if (!content.isLoading) {
+      content.load()
+    }
   }, [ui.selectedProject])
 
-  if (content.isLoaded)
+  const contentList = content.list()
+  if (content.isLoading) {
     return (
       <LogsExplorerLayout>
-        <Loading active={true}>loading</Loading>
+        <Loading active={true}>{null}</Loading>
       </LogsExplorerLayout>
     )
+  }
 
-  const saved = content.list()
-
+  const saved: UserContent[] = (contentList?.[0] || []).filter((content: UserContent)=> content.type === 'log_sql')
```

Comment:
possibly could deal with this in the store itsself.
could just refactor later.



```tsx
  // studio/stores/content/ProjectContentStore.ts
  
  // add this
  logsQueries(filter?: any) {
    let arr = Object.values(this.data)
    if (!!filter) {
      const arr_filtered = arr
        .filter((x) => x.type == 'log_sql')
        .filter(filter)
        .sort((a, b) => a.name.localeCompare(b.name))

      return arr_filtered
    } else {
      const arr_filtered = arr
        .filter((x) => x.type == 'log_sql')
        .sort((a, b) => a.name.localeCompare(b.name))

      return arr_filtered
    }
  }
  
 // in page
 const {content} = useStore()
 await content.load()
 const queries = content.logsQueries()
  ```

Reply:
should I wait for @phamhieu 's https://github.com/supabase/supabase/pull/6586 ?